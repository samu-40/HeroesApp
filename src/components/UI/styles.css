.navbar{
    width: 100%;
    height: 3rem;
    display: flex;
    align-items: center;
    padding: 0 13px;
    background: #222;
}

.brand{
    text-decoration: none;
    color: #fff;
    font-size: 1.3rem;
}

.links{
    margin-left: 2.5rem;
    flex-grow: 1;
}

.link{
    text-decoration: none;
    color: #999;
    margin-right: 1rem;
    padding: 2px;
    transition: color .2s;
}

.link:hover{
    color: #fff;
}

.active{
    color: #fff;
}

span{
    margin-right: 10px;
    color: #44f;
}

button{
    border: none;
    background: none;
    font-size: 1rem;
    color: #999;
    padding: 2px;
    transition: color .2s;
}

button:hover{
    color: #fff;
}