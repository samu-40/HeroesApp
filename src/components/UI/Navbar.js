import React, { useContext } from 'react';
import { Link, NavLink, useHistory } from 'react-router-dom';
import { AuthContext } from '../../auth/AuthContext';
import { types } from '../../types/types';

import './styles.css';

export const Navbar = () => {

    const {user, dispatch} = useContext(AuthContext);

    const history = useHistory();

    const handleLogout = () => {
        history.replace('/login');

        dispatch({
            type: types.logout
        });
    };

    return (
        <nav className='navbar'>
            <Link
                className="brand"
                to="/"
            >
                Asociaciones
            </Link>

            <div className='links'>
                <NavLink
                    activeClassName="active"
                    className="link"
                    exact
                    to="/marvel"
                >
                    Marvel
                </NavLink>

                <NavLink
                    activeClassName="active"
                    className="link"
                    exact
                    to="/dc"
                >
                    DC
                </NavLink>
                <NavLink
                    activeClassName="active"
                    className="link"
                    exact
                    to="/search"
                >
                    Search
                </NavLink>
            </div>
            <div>
                <span>
                    {user.name}
                </span>
                <button
                    onClick={handleLogout}
                >
                    Logout
                </button>
            </div>
        </nav>
    )
}