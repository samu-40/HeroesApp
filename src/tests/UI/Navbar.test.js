import '../../setupTest';
import { mount } from "enzyme";
import { AuthContext } from '../../auth/AuthContext';
import { Navbar } from '../../components/UI/Navbar';
import { MemoryRouter, Router } from 'react-router-dom';
import { types } from "../../types/types";

describe('Pruebas en Navbar.js', () => {

    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged: true,
            name: 'Samuel'
        }
    };

    const historyMock = {
        push: jest.fn(),
        replace: jest.fn(),
        location: {},
        listen: jest.fn(),
        createHref: jest.fn()
    };

    const wrapper = mount(
        <AuthContext.Provider value={contextValue}>
            <MemoryRouter>
                <Router history={historyMock}>
                    <Navbar/>
                </Router>
            </MemoryRouter>
        </AuthContext.Provider>
    );


    afterEach(() => {
        jest.clearAllMocks();
    });

    test('Mostrase correctamente ', () => {
        
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('span').text().trim()).toBe(contextValue.user.name);

    });

    test('Debe llamar el logout y llamar al history', () => {
        
        wrapper.find('button').prop('onClick')();

        expect(contextValue.dispatch).toHaveBeenCalledWith({
            type: types.logout
        });

        expect(historyMock.replace).toHaveBeenCalledWith('/login');

    });    
    
});
