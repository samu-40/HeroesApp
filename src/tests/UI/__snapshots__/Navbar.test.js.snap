// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en Navbar.js Mostrase correctamente  1`] = `
<nav
  className="navbar"
>
  <a
    className="brand"
    onClick={[Function]}
  >
    Asociaciones
  </a>
  <div
    className="links"
  >
    <a
      aria-current={null}
      className="link"
      onClick={[Function]}
    >
      Marvel
    </a>
    <a
      aria-current={null}
      className="link"
      onClick={[Function]}
    >
      DC
    </a>
    <a
      aria-current={null}
      className="link"
      onClick={[Function]}
    >
      Search
    </a>
  </div>
  <div>
    <span>
      Samuel
    </span>
    <button
      onClick={[Function]}
    >
      Logout
    </button>
  </div>
</nav>
`;
